astropy/coordinates/earth.py

```diff
@@ -368,13 +368,33 @@ def to_geocentric(self):
     def itrs(self):
         """
         Generates an `~astropy.coordinates.ITRS` object with the coordinates of
-        this object.
+        this object at the default ``obstime``.
         """
-        #potential circular imports prevent this from being up top
+        #circular imports prevent this from being up top
         from .builtin_frames import ITRS
 
         return ITRS(x=self.x, y=self.y, z=self.z)
 
+    def get_itrs(self, obstime):
+        """
+        Generates an `~astropy.coordinates.ITRS` object with the coordinates of
+        this object at the requested ``obstime``.
+
+        Parameters
+        ----------
+        obstime : `~astropy.time.Time`
+            The ``obstime`` to apply to the new `~astropy.coordinates.ITRS`
+
+        Returns
+        -------
+        itrs : `~astropy.coordinates.ITRS`
+            The new object in the ITRS frame
+        """
+        #circular imports prevent this from being up top
+        from .builtin_frames import ITRS
```

Comment:
That alone didn't work, but it turns out an alternative solution was to move the import of `EarthLocation` in `baseframe` to the bottom.  Will push up a commit with that shortly.